tests/runtest.sh

```diff
@@ -245,39 +254,38 @@ CoreRT_ExtraCXXFlags=
 CoreRT_ExtraLinkFlags=
 if [ ${CoreRT_CrossBuild} != 0 ]; then
     CoreRT_TestRun=false
-    CoreRT_CrossRootFS=${CoreRT_TestRoot}/../cross/rootfs/${CoreRT_BuildArch}
     # all values are brought from the appropriate toolchain.cmake's
     case $CoreRT_BuildArch in
          arm)
-             CoreRT_CrossCXXFlags="-target armv7-linux-gnueabihf -mthumb -mfpu=vfpv3 --sysroot=${CoreRT_CrossRootFS}"
-             CoreRT_CrossLinkerFlags="-target arm-linux-gnueabihf -B ${CoreRT_CrossRootFS}/usr/lib/gcc/arm-linux-gnueabihf `
-                                     `-L${CorRT_CrossRootFS}/lib/arm-linux-gnueabihf --sysroot=${CoreRT_CrossRootFS}"
+             CoreRT_CrossCXXFlags="-target armv7-linux-gnueabihf -mthumb -mfpu=vfpv3 --sysroot=${ROOTFS_DIR}"
+             CoreRT_CrossLinkerFlags="-target arm-linux-gnueabihf -B ${ROOTFS_DIR}/usr/lib/gcc/arm-linux-gnueabihf `
+                                     `-L${CorRT_CrossRootFS}/lib/arm-linux-gnueabihf --sysroot=${ROOTFS_DIR}"
          ;;
          arm64)
-             CoreRT_CrossCXXFlags="-target aarch64-linux-gnu --sysroot=${CoreRT_CrossRootFS}"
-             CoreRT_CrossLinkerFlags="-target aarch64-linux-gnu -B ${CoreRT_CrossRootFS}/usr/lib/gcc/aarch64-linux-gnu `
-                                     `-L${CoreRT_CrossRootFS}/lib/aarch64-linux-gnu --sysroot=${CoreRT_CrossRootFS}"
+             CoreRT_CrossCXXFlags="-target aarch64-linux-gnu --sysroot=${ROOTFS_DIR}"
+             CoreRT_CrossLinkerFlags="-target aarch64-linux-gnu -B ${ROOTFS_DIR}/usr/lib/gcc/aarch64-linux-gnu `
+                                     `-L${ROOTFS_DIR}/lib/aarch64-linux-gnu --sysroot=${ROOTFS_DIR}"
          ;;
          armel)
-             CoreRT_CrossCXXFlags="-target armv7-linux-gnueabi -mthumb -mfpu=vfpv3  -mfloat-abi=softfp --sysroot=${CoreRT_CrossRootFS}"
-             CoreRT_CrossLinkerFlags="-target arm-linux-gnueabi --sysroot=${CoreRT_CrossRootFS}"
-             ID=
-             if [ -e $ROOTFS_DIR/etc/os-release ]; then
-                 source $ROOTFS_DIR/etc/os-release
-             fi
-             if [ "$ID" = "tizen" ]; then
+             CoreRT_CrossCXXFlags="-target armv7-linux-gnueabi -mthumb -mfpu=vfpv3  -mfloat-abi=softfp --sysroot=${ROOTFS_DIR}"
+             CoreRT_CrossLinkerFlags="-target arm-linux-gnueabi --sysroot=${ROOTFS_DIR}"
+             if [ "$DISTRO_ID" = "tizen" ]; then
+                 # native executing is not working in release case for Tizen ARM and Legacy Jit now
```

Comment:
From what I have seen on my Raspberry Pi, building with optimization off has problem as well - you just tend to be  lucky to survive the stack corruption caused by the problem.

Reply:
Maybe, in any case, we will work with this and fix it later on the RyuJIT version.